### PR TITLE
Translate blueprints to pipelines

### PR DESCRIPTION
**Pull request by teg**
Add basic infrastructure for blueprint to pipeline translations.

### CODE REVIEW

File: internal/blueprint/blueprint.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Package blueprint contains primites for representing weldr blueprints and
```
Comment (larskarlitski) [inline internal/blueprint/blueprint.go]:
```suggestion
// Package blueprint contains primitives for representing weldr blueprints and
```

File: internal/blueprint/blueprint.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import "osbuild-composer/internal/pipeline"
+
+// A Blueprint is a high-level description of an image.
+type Blueprint struct {
+	Name        string    `json:"name"`
+	Description string    `json:"description"`
+	Version     string    `json:"version,omitempty"`
+	Packages    []Package `json:"packages"`
+	Modules     []Package `json:"modules"`
+	Groups      []Package `json:"groups"`
```
Comment (larskarlitski) [inline internal/blueprint/blueprint.go]: Maybe we should only have one thing here? Weldr has all three, but puts packages in each of them. We'd need to support this still in the API, but could combine them here.

File: internal/blueprint/blueprint.go [hunk 3]
```diff
@@ -20,9 +23,34 @@ type Package struct {
 	Version string `json:"version,omitempty"`
 }
 
+type translator interface {
+	translate(b *Blueprint) *pipeline.Pipeline
+}
```
Comment (larskarlitski) [inline internal/blueprint/blueprint.go]: This type is called `translator`, but the files are `XXX_output.go`. Does it make sense to give those consistent names?

Comment (teg) [reply internal/blueprint/blueprint.go]: Yeah, I'll rename them all "outputs".

File: internal/blueprint/blueprint.go (hunk 2 above)
Comment (teg) [reply internal/blueprint/blueprint.go]: Hm, let's address that in a follow-up? (I would have to look into how it works, this currently just preserves the behaviour).

**Comment by teg** [comment]
Pushed an update, also added accessors for filename and mimetype per output type. This in order to avoid hard-coding those things in more than one place.

File: internal/blueprint/blueprint.go (hunk 2 above)
Comment (larskarlitski) [reply internal/blueprint/blueprint.go]: Yeah, I was just thinking out loud.

**Pull request merged**